pkg/game/board.go

```diff
@@ -0,0 +1,126 @@
+package game
+
+import (
+	"errors"
+	"slices"
+
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/game/elements"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles"
+)
+
+// mutable type
+// Position coordinates example on the board:
+// (-1, +1)  (+0, +1)  (+1, +1)
+// (-1, +0)  (+0, +0)  (+1, +0)
+// (-1, -1)  (+0, -1)  (+1, -1)
+//
+// Starting tile is placed at (+0, +0) position.
+type board struct {
+	tileSet []tiles.Tile
+	// The information about the tile and its placement is stored sparsely
+	// in a slice of size equal to the number of tiles in the set.
+	// `tiles[0]` is always the starting tile.
+	// Indexes in `tiles[1:]` correspond to positions of equivalent tiles in `tileSet`.
+	tiles []elements.PlacedTile
+	// tilesMap is used by the engine for faster lookups
+	// but contains the same information as the `tiles` slice.
+	tilesMap map[elements.Position]elements.PlacedTile
+}
+
+func NewBoard(tileSet []tiles.Tile) elements.Board {
```

Comment:
Consider adding starting tile as a parameter. It will be usefull in the future, when we implement River expansion (if a starting tile is a source then we add river tiles).